### PR TITLE
absolute_url should not mangle URL if called more than once

### LINKED ISSUE #5788: absolute_url mangles URL if called twice
- [x] I believe this to be a bug, not a question about using Jekyll.
- [x] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [x] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [ ] This is a feature request.

---

- [ ] I was trying to install.
- [ ] There is a broken Plugin API.
- [ ] I had an error on GitHub Pages, and I have reproduced it locally.
- [ ] I had an error on GitHub Pages, and GitHub Support said it was a Jekyll Bug.
- [ ] I had an error on GitHub Pages and I did not test it locally.
- [ ] I was trying to build.
- [x] It was another bug.

## My Reproduction Steps

```text
{{ '/' | absolute_url | absolute_url }}
```

Yields `http://example.com/http://example.com/`

## The Output I Wanted

`http://example.com/`

The filter should check to see if the URL is already absolute, and leave it alone if it is.

/cc @jekyll/core

### PR DESCRIPTION
**Pull request by jekyllbot**
PR automatically created for @pathawks.

```text
Failing test: absolute_url with absolute URL
```

Fixes #5788

### CODE REVIEW

**Comment by pathawks** [comment]
Thinking about this a bit more, perhaps we should issue a debug/warning message instead of / in addition to passing input through unchanged.

**Comment by parkr** [comment]
> Thinking about this a bit more, perhaps we should issue a debug/warning message instead of / in addition to passing input through unchanged.

How come?

**Comment by pathawks** [comment]
#### Why issue a debug message?

I came across this when I was working on a plugin. I assigned an `absolute_url` to a variable, and then I filtered that variable through `absolute_url` when I actually output it. These lines were not very close in proximity, so I only noticed my error when the test failed.

I certainly wouldn't want to ship code like this, and I'm sure any plugin or theme author worth her salt would agree. _So_, it would be nice to get a message that this is happening.

#### Why allow the URL to pass through?

```rib
2.3.0 :001 > require "Addressable"
 => true 
2.3.0 :002 > Addressable::URI.parse("http://example.com/http://example.com").to_s
 => "http://example.com/http://example.com"
```

Addressable seems to be just fine with `http://example.com/http://example.com` as a URI, and I don't want to end up on one of those **Falsehoods Programmers Believe About&hellip;** lists.

See also: #5284 #5480

**Comment by parkr** [comment]
@pathawks OK, how would you like to proceed?

**Comment by pathawks** [comment]
I think it should just output a warning, but _I guess_ it should go head and build the URL as requested.

Am I doing this right? I'm not sure how to write a test to make sure it outputs a warning.

File: lib/jekyll/filters/url_filters.rb [hunk 1]
```diff
@@ -10,6 +10,12 @@ module URLFilters
       # Returns the absolute URL as a String.
       def absolute_url(input)
         return if input.nil?
+        if Addressable::URI.parse(input).absolute?
+          Jekyll.logger.warn(
+            "absolute_url:",
+              %("#{input}" appears to already be absolute)
+          )
```
Comment (parkr) [inline lib/jekyll/filters/url_filters.rb]: From a usability standpoint, the first thing I think about is: how is the user going to have any idea *where* in their site to look? If you don't think it should be automatically dealt with, then we should `raise` or something so we can get some context.

**Comment by parkr** [comment]
@pathawks You cool with my suggestion above?

**Comment by pathawks** [comment]
@parkr Yessir, how does it look now?

**Comment by parkr** [comment]
@jekyllbot: merge +bug

**Pull request merged**